tests/regression/roles/sonic_pki/defaults/main.yml

```diff
@@ -0,0 +1,100 @@
+---
+ansible_connection: httpapi
+module_name: tacacs_pki
+
+preparations_tests:
+  delete_pki:
+    - name: delete_all
+      description: Delete PKI
+      state: deleted
+      input:
+tests:
+  - name: test_case_01
+    description: Add new PKI configuration
+    state: merged
+    input:
+      security_profiles:
+        - profile_name: mysp
+        - profile_name: myspb
+          ocsp_responder_list:
+            - http://example.com/ocsp
+      trust_stores:
+        - name: myts
+          ca_name: CA2
+
+  - name: test_case_02
+    description: Update specific params of security-profile
+    state: merged
+    input:
+      security_profiles:
+        - profile_name: mysp
+          cdp_list:
+            - http://example.com/cdp
+          trust_store: myts
+          revocation_check: false
+  - name: test_case_03
+    description: Delete specific params of security-profile
+    state: deleted
+    input:
+      security_profiles:
+        - profile_name: mysp
+          cdp_list:
+            - http://example.com/cdp
+  - name: test_case_04
+    description: Delete all security_profiles configurations
+    state: deleted
+    input:
+      security_profiles:
+        - profile_name: mysp
+        - profile_name: myspb
+
+  - name: test_case_05
+    description: Merge parameter of security_profiles configurations
+    state: merged
+    input:
+      security_profiles:
+        - profile_name: mysp
+          revocation_check: false
+          peer_name_check: true
+        - profile_name: myspb
+          ocsp_responder_list:
+            - http://example.com/ocsp
+          trust_store: myts
+          key_usage_check: true
+
+  - name: test_case_06
```

Comment:
This is a good test case, but the code needs to be changed to handle it in a way that replaces the entire security profile with the new content. (A corresponding comment has been posted inline in the "config" file diffs.)